cmake/compiler/msvc.cmake

```diff
@@ -11,7 +11,7 @@
 
 
 if(NOT CRT_OPT)
-  set(CRT_OPT /MD)
+  set(CRT_OPT $<$<CONFIG:Release>:/MD>)
```

Comment:
We have to always use `/MD`, or else build different versions of our 3rd party dependencies like Boost and PyTorch.

The standard Windows way of doing things would be to use `/MDd` when doing a debug build, but that would require linking against Boost, PyTorch, etc, that were also built with `/MDd`.  It would be a lot of work to build two sets of all our 3rd party dependencies, especially given that we almost never debug anything on Windows.

So it would be better if instead of doing this we could tell CMake that we want it to use `/MD` instead of `/MDd` for debug builds on Windows.  Or, if that's too hard, just revert this line as the rest of the PR is making things better than they were.

Reply:
Ah ok, that makes sense - I'll back that change out.